src/opm/input/eclipse/Schedule/Well/Connection.cpp

```diff
@@ -443,4 +444,59 @@ Connection::CTFKind Connection::kind() const {
     return m_ctfkind;
 }
 
+Connection::FilterCakeGeometry
+Connection::filterCakeGeometryFromString(const std::string& str)
+{
+    if (str == "LINEAR")
+        return Connection::FilterCakeGeometry::LINEAR;
+    else if (str == "RADIAL")
+        return Connection::FilterCakeGeometry::RADIAL;
+    else if (str == "NONE")
+        return Connection::FilterCakeGeometry::NONE;
+    else
+        throw std::invalid_argument("Unknow enum INJMultMode string: " + str);
+}
+
+void Connection::setFilterCake(const Connection::FilterCake& filter_cake) {
+        this->m_filter_cake = filter_cake;
 }
+
+
+const Connection::FilterCake& Connection::getFilterCake() const {
+        return this->m_filter_cake;
+}
+
+
+double Connection::getFilterCakeRadius() const {
+    if (this->m_filter_cake.radius) {
+        return *this->m_filter_cake.radius;
+    } else {
+        return this->m_rw;
+    }
+}
+
+
+double Connection::getFilterCakeArea() const {
+    if (this->m_filter_cake.flow_area) {
+        return *(this->m_filter_cake.flow_area);
+    } else {
+        const double radius = this->getFilterCakeRadius();
+        const double length = this->m_connection_length;
+        constexpr double pi = 3.14159265;
+        return 2. * pi * radius * length;
+    }
+}
+
+Connection::FilterCake::FilterCake(const DeckRecord& record) {
+    this->geometry = Connection::filterCakeGeometryFromString(record.getItem("GEOMETRY").getTrimmedString(0));
+    this->perm = record.getItem("FILTER_CAKE_PERM").getSIDouble(0);
+    this->poro = record.getItem("FILTER_CAKE_PORO").getSIDouble(0);
+    if (!record.getItem("FILTER_CAKE_RADIUS").defaultApplied(0)) {
+        this->radius = record.getItem("FILTER_CAKE_RADIUS").getSIDouble(0);
+    }
+    if (!record.getItem("FILTER_CAKE_AREA").defaultApplied(0)) {
+        this->flow_area = record.getItem("FILTER_CAKE_AREA").getSIDouble(0);
+    }
```

Comment:
I'd prefer it if you'd use the compiled item names here please.  For example
```C++
this->perm = record.getItem<ParserKeywords::WINJDAM::FILTER_CAKE_PERM>().getSIDouble(0);
```
That way we get a compile-time error if the parameter name is spelled incorrectly.  You must
```C++
#include <opm/input/eclipse/Parser/ParserKeywords/W.hpp>
```
to enable using the compiled item names though.